backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidZipFileHandle.java

```diff
@@ -0,0 +1,109 @@
+/*******************************************************************************
+ * Copyright 2011 See AUTHORS file.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.badlogic.gdx.backends.android;
+
+import java.io.*;
+
+import android.content.res.AssetFileDescriptor;
+
+import com.badlogic.gdx.Files.FileType;
+import com.badlogic.gdx.files.FileHandle;
+
+/** @author sarkanyi */
+public class AndroidZipFileHandle extends AndroidFileHandle {
+	private static class AudioExtensions {
+		private static AudioExtensions instance;
+		private ObjectSet<String> extensions;
+
+		private AudioExtensions() {
+			extensions = new ObjectSet<String>();
+			extensions.addAll(".mp3");
+			extensions.addAll(".wma");
+			extensions.addAll(".wav");
+			extensions.addAll(".mp2");
+			extensions.addAll(".aac");
+			extensions.addAll(".ac3");
+			extensions.addAll(".au");
+			extensions.addAll(".ogg");
+			extensions.addAll(".flac");
+		}
+		static private AudioExtensions getInstance() {
+			if(instance == null)
+				instance = new AudioExtensions();
+			return instance;
+		}
+		public static boolean isAudio(String fileName) {
+			if(fileName.contains(".")) {
+				String ext = fileName.substring(fileName.lastIndexOf("."), fileName.length());
+				return getInstance().extensions.contains(ext.toLowerCase(Locale.ENGLISH));
+			}
+			return false;
+		}
+	}
+
+	private AssetFileDescriptor assetFd;
+
+	public AndroidZipFileHandle(String fileName) {
+		super(null, fileName, AudioExtensions.isAudio(fileName) ? FileType.Internal : FileType.Absolute);
```

Comment:
Because it works that way :). I don't know exactly why, but it's tested and works. I missed the imports because I was adapting the stuff in my game and testing the changes, but then I moved only the new class/function.
If I think about it long and hard, I think it's actually a "hack", because this way the AndroidFileHandle.assets will be null for audio if it's not internal, enabling it to enter the second codepath which returns the actual AssetFileDescriptor. Android's AssetManager can't open it with openFd() to get the AssetFileDescriptor. That's certain.